src/Symfony/Component/Security/Http/EventListener/IsGrantedAttributeListener.php

```diff
@@ -54,20 +54,26 @@ public function onKernelControllerArguments(ControllerArgumentsEvent $event): vo
                     foreach ($subjectRef as $refKey => $ref) {
                         $subject[\is_string($refKey) ? $refKey : (string) $ref] = $this->getIsGrantedSubject($ref, $request, $arguments);
                     }
+                } elseif ($subjectRef instanceof \Closure) {
+                    $subject = $subjectRef($request, $arguments);
```

Comment:
should we pass arguments first, for cases where the closure needs only arguments ?

Reply:
Updated as well, it makes more sense to passe args first as they're more likely to be used!